app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeModalActivity.java

```diff
@@ -0,0 +1,50 @@
+package com.sdpteam.connectout.qr_code;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.sdpteam.connectout.R;
+
+public class QRcodeModalActivity extends AppCompatActivity {
+
+    private ImageView qrCodeImageView;
+    private TextView modalTitle;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_qrcode_modal);
+
+        qrCodeImageView = findViewById(R.id.qrcode_image);
+        modalTitle = findViewById(R.id.modal_title);
+
+        // retrieving data from intent
+        String title = getIntent().getStringExtra("title");
+        String qrCodeData = getIntent().getStringExtra("qrCodeData");
+
+        // displaying the title and the qr code
+        modalTitle.setText(title);
+        QRcodeGenerator qrCode = new QRcodeGenerator(); // any other way ?
```

Comment:
looks good to me... what other way are you thinking of ?

Reply:
nothing much, I initially thought about making this class static but it's alright now